src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -372,6 +382,83 @@ private ClientId getClientIdForSigningCert(String instanceIdentifier, X509Certif
         return certificateSubject;
     }
 
+    /**
+     * Send the authentication certificate registration request to central server
+     * @param hash certificate hash
+     * @param securityServerAddress IP address or DNS name of the security server
+     * @throws CertificateNotFoundException
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     */
+    public void registerAuthCert(String hash, String securityServerAddress) throws CertificateNotFoundException,
+            GlobalConfService.GlobalConfOutdatedException, InvalidCertificateException,
+            SignCertificateNotSupportedException {
+        CertificateInfo certificateInfo = getCertificateInfo(hash);
+        verifyAuthCert(certificateInfo);
+        SecurityServerId securityServerId = serverConfService.getSecurityServerId();
+        try {
+            managementRequestService.sendAuthCertRegisterRequest(securityServerId, securityServerAddress,
+                    certificateInfo.getCertificateBytes());
+            signerProxyFacade.setCertStatus(certificateInfo.getId(), CertificateInfo.STATUS_REGINPROG);
+        } catch (GlobalConfService.GlobalConfOutdatedException e) {
+            throw e;
+        } catch (Exception e) {
+            if (e instanceof CodedException) {
+                CodedException ce = (CodedException) e;
+                if (isCausedByCertNotFound(ce)) {
+                    throw new CertificateNotFoundException(ce);
+                }
+                throw ce;
+            }
+            throw new RuntimeException("Could not set certificate status", e);
+        }
+    }
+
+    /**
+     * Send the authentication certificate deletion request to central server
+     * @param hash certificate hash
+     * @throws CertificateNotFoundException
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     */
+    public void unregisterAuthCert(String hash) throws CertificateNotFoundException,
```

Comment:
unregisterAuthCert and registerAuthCert are mostly identical, so it _might_ make sense to share code instead of duplicating the logic? I am not 100% sure though.

Reply:
I looked into this approach but the outcome was IMO a little too complex compared to the small amount of duplicate code in it now.